app/src/androidTest/java/com/github/se/studybuddies/tests/CreateToDoTest.kt

```diff
@@ -32,80 +39,199 @@ class CreateToDoTest : TestCase(kaspressoBuilder = Kaspresso.Builder.withCompose
+        assertIsDisplayed()
+      }
+
+      topAppBar {
+        // arrange: verify pre-conditions
+        assertIsDisplayed()
+      }
+      divider {
+        // arrange: verify pre-conditions
+        assertIsDisplayed()
+      }
+      goBackButton {
+        // arrange: verify pre-conditions
+        assertIsDisplayed()
+        performClick()
+      }
+    }
+    // assert: the nav action has been called
+    verify { mockNavActions.navigateTo(Route.TODOLIST) }
+    confirmVerified(mockNavActions)
+  }
+
+  @Test
+  fun inputTaskName() {
+    onComposeScreen<CreateToDoScreen>(composeTestRule) {
+      saveButton { assertIsNotEnabled() }
+      todoNameField {
+        performTextClearance()
+        performTextInput("Official Task Testing")
+        assertTextContains("Official Task Testing")
+      }
+      ViewActions.closeSoftKeyboard()
+      saveButton {
+        assertIsEnabled()
+        performClick()
+      }
+      verify { mockNavActions.goBack() }
+      confirmVerified(mockNavActions)
+    }
+  }
+
+  @Test
+  fun inputDescription() {
+    onComposeScreen<CreateToDoScreen>(composeTestRule) {
+      todoNameField {
+        performTextClearance()
+        performTextInput("Official Task Testing")
+      }
+      todoDescriptionField {
+        performTextClearance()
+        performTextInput("Official Task Testing")
+        assertTextContains("Official Task Testing")
+      }
+      ViewActions.closeSoftKeyboard()
+      saveButton {
+        assertIsEnabled()
+        performClick()
+      }
+      verify { mockNavActions.goBack() }
+      confirmVerified(mockNavActions)
+    }
+  }
+
+  @Test
+  fun datePickerTest() {
+    onComposeScreen<CreateToDoScreen>(composeTestRule) {
+      todoDateField { performClick() }
+      datePicker { assertIsDisplayed() }
+      dateConfirmButton {
+        assertIsDisplayed()
+        assertHasClickAction()
+        assertTextEquals(("Confirm"))
+      }
+      dateDismissButton {
+        assertIsDisplayed()
+        assertHasClickAction()
+        assertTextEquals(("Cancel"))
+        performClick()
+      }
+      datePicker { assertIsNotDisplayed() }
+    }
+  }
 
   /*
-      @Test
-      fun topAppBarTest() = run {
-        onComposeScreen<CreateToDoScreen>(composeTestRule) {
-          topAppBox {
-            // arrange: verify pre-conditions
-            assertIsDisplayed()
-          }
-          topAppBar {
-            // arrange: verify pre-conditions
-            assertIsDisplayed()
-          }
-          divider {
-            // arrange: verify pre-conditions
-            assertIsDisplayed()
-          }
-          goBackButton {
-            // arrange: verify pre-conditions
-            assertIsDisplayed()
-            performClick()
-          }
-        }
-        // assert: the nav action has been called
-        verify { mockNavActions.navigateTo(Route.TODOLIST) }
-        confirmVerified(mockNavActions)
+  @Test
+  fun inputDate() {

```

Comment:
Not sure if you forgot to delete this commented out code or if the test isn't passing the CI yet ?

Reply:
It has not passed CI yet, i will try to solve this later if given the time